tests/gradients/test_parameter_shift.py

```diff
@@ -1273,11 +2087,193 @@ def cost_fn(x):
+                np.sin(x / 2) ** 2 * np.cos(y / 2) ** 2,
+            ]
+        ]
+    )
+    expected_first_order = np.array(
+        [
+            [[-np.sin(x) * np.cos(y / 2) ** 2 / 2, -np.sin(y) * np.cos(x / 2) ** 2 / 2]],
+            [[-np.sin(x) * np.sin(y / 2) ** 2 / 2, np.sin(y) * np.cos(x / 2) ** 2 / 2]],
+            [[np.sin(x) * np.sin(y / 2) ** 2 / 2, np.sin(y) * np.sin(x / 2) ** 2 / 2]],
+            [[np.sin(x) * np.cos(y / 2) ** 2 / 2, -np.sin(y) * np.sin(x / 2) ** 2 / 2]],
+        ]
+    )
+    expected_second_order = np.array(
+        [
+            [
+                [
+                    [-np.cos(x) * np.cos(y / 2) ** 2 / 2, np.sin(y) * np.sin(x) / 4],
+                    [np.sin(y) * np.sin(x) / 4, -np.cos(y) * np.cos(x / 2) ** 2 / 2],
+                ]
+            ],
+            [
+                [
+                    [-np.cos(x) * np.sin(y / 2) ** 2 / 2, -np.sin(y) * np.sin(x) / 4],
+                    [-np.sin(y) * np.sin(x) / 4, np.cos(y) * np.cos(x / 2) ** 2 / 2],
+                ]
+            ],
+            [
+                [
+                    [np.cos(x) * np.sin(y / 2) ** 2 / 2, np.sin(y) * np.sin(x) / 4],
+                    [np.sin(y) * np.sin(x) / 4, np.cos(y) * np.sin(x / 2) ** 2 / 2],
+                ]
+            ],
+            [
+                [
+                    [np.cos(x) * np.cos(y / 2) ** 2 / 2, -np.sin(y) * np.sin(x) / 4],
+                    [-np.sin(y) * np.sin(x) / 4, -np.cos(y) * np.sin(x / 2) ** 2 / 2],
+                ]
+            ],
+        ]
+    )
+
+    def make_tape(self, x):
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(x[0], wires=[0])
+            qml.RY(x[1], wires=[1])
+            qml.CNOT(wires=[0, 1])
+            qml.probs(wires=[0, 1])
+        return tape
+
+    @pytest.mark.autograd
+    def test_autograd(self, tol, broadcast, expected):
+        """Tests that the output of the parameter-shift transform
+        can be differentiated using autograd, yielding second derivatives."""
+        dev = qml.device("default.qubit.autograd", wires=2)
+        params = np.array([0.543, -0.654], requires_grad=True)
+        exp_num_tapes, exp_batch_sizes = expected
+
+        res = dev.execute(self.make_tape(params))
+        assert np.allclose(res, self.expected_res)
+
+        def cost_fn(x):
+            tape = self.make_tape(x)
+            tape.trainable_params = {0, 1}
+            tapes, fn = qml.gradients.param_shift(tape, broadcast=broadcast)
+            assert len(tapes) == exp_num_tapes
+            assert [t.batch_size for t in tapes] == exp_batch_sizes
+            jac = fn(dev.batch_execute(tapes))
+            return jac
+
+        res = cost_fn(params)
+        assert np.allclose(res, self.expected_first_order, atol=tol, rtol=0)
+
+        res = qml.jacobian(cost_fn)(params)
+        assert np.allclose(res, self.expected_second_order, atol=tol, rtol=0)
+
+    @pytest.mark.tf
+    @pytest.mark.slow

```

Comment:
Just curious: how slow is this test? Also, would any other test be similarly slow?